src/components/select/filterable-select/filterable-select.component.js

```diff
@@ -77,6 +77,10 @@ const FilterableSelect = React.forwardRef(
           const match = findElementWithMatchingText(newFilterText, children);
           const isFilterCleared = isDeleteEvent && newFilterText === "";
 
+          if ((!match || isFilterCleared) && onChange) {
+            onChange(createCustomEvent(""));
+          }
+
```

Comment:
to avoid code repeating you could create a method and reuse it:

```
    function triggerChange(newValue) {
      if (onChange) {
        onChange(createCustomEvent(newValue));
      }
    }
```

Reply:
@grabkowski good point, thanks, changed